### ISSUE TITLE
How to future-proof database/avoid migration with version changes?

### ISSUE THREAD

**Issue by devmaestro1** [litedb-org/LiteDB#2179]
Hello,

I am wondering what is the best approach to ensure that I minimize the necessity of database migrations as my code changes.
In particular, I note that LiteDB stores the fully-qualified class name along with the version number. Does this mean that if I increment my assembly version number I will break existing data? If so, is there any way around this so it uses only the class name and not version (etc)?

Thanks.

**Comment by JKamsker** [comment]
No, the assembly version isn't stored, so bumping it is safe. `_type` is `"Namespace.TypeName, AssemblyName"`, with no version, culture or public key token. It is written only when the runtime type differs from the declared type, for example a `Dog` in an `Animal` property, or any value in an `object` or interface property. Ordinary POCO properties get no `_type` at all.

What does break `_type` values is renaming the namespace, the class or the assembly. To be independent of those names, plug in your own `ITypeNameBinder` with stable aliases. It can also map old names to renamed types:

```csharp
public class AliasBinder : ITypeNameBinder
{
    static readonly Dictionary<string, Type> ByName = new() { ["dog"] = typeof(Dog), ["OldNs.Dog, OldAsm"] = typeof(Dog) };
    static readonly Dictionary<Type, string> ByType = new() { [typeof(Dog)] = "dog" };

    public string GetName(Type t) => ByType.TryGetValue(t, out var n) ? n : DefaultTypeNameBinder.Instance.GetName(t);
    public Type GetType(string n) => ByName.TryGetValue(n, out var t) ? t : DefaultTypeNameBinder.Instance.GetType(n);
}

var db = new LiteDatabase("app.db", new BsonMapper(typeNameBinder: new AliasBinder()));
```

Other class changes are tolerant: fields your class no longer has are ignored, and new properties keep their default values. `[BsonField("OldName")]` handles property renames, and `db.UserVersion` can track your own schema version if you do need a migration step.

Closing this as answered — if it doesn't cover your case, comment here and we'll reopen it.
